### PR TITLE
refactor: begin llmq deglobalisation

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
This does some of the easy deblogalisation in LLMQ code. There is much more work to do in this regard, but this PR is a good first step. The biggest accomplishment I guess of this PR is fully removing the `quorumDKGDebugManager` global. However, we also changed a lot of global member accesses into class member accesses.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1911,7 +1912,7 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     }
 #endif
 
-    pdsNotificationInterface = new CDSNotificationInterface(*node.connman);
+    pdsNotificationInterface = new CDSNotificationInterface(*node.connman, *llmq::quorumInstantSendManager);
```
Comment (UdjinM6) [inline src/init.cpp:1915]: can't do this bea52702d045a25ab626082deada21c850fe07e6

File: src/init.cpp [hunk 2]
```diff
@@ -2003,7 +2004,7 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
                 llmq::quorumSnapshotManager.reset();
                 llmq::quorumSnapshotManager.reset(new llmq::CQuorumSnapshotManager(*evoDb));
 
-                llmq::InitLLMQSystem(*evoDb, *node.mempool, *node.connman, false, fReset || fReindexChainState);
+                llmq::InitLLMQSystem(node, *evoDb, *node.mempool, *node.connman, false, fReset || fReindexChainState);
```
Comment (UdjinM6) [inline src/init.cpp:2007]: can be simplified now a332f39184079c0a5b54bff3953888971316bcc9

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/signing.h [hunk 3]
```diff
@@ -178,7 +183,7 @@ class CSigningManager
     std::vector<CRecoveredSigsListener*> recoveredSigsListeners GUARDED_BY(cs);
 
 public:
-    CSigningManager(CConnman& _connman, bool fMemory, bool fWipe);
+    CSigningManager(CConnman& _connman, CQuorumManager& quorumMan, CSigSharesManager& sigSharesMan, bool fMemory, bool fWipe);
```
Comment (kittywhiskers) [inline src/llmq/signing.h:186]: nit: convention usually means they're named `_quorumManager` and `_sigSharesManager`

File: src/llmq/instantsend.cpp [hunk 4]
```diff
@@ -27,8 +27,8 @@
 namespace llmq
 {
 
-static const std::string INPUTLOCK_REQUESTID_PREFIX = "inlock";
-static const std::string ISLOCK_REQUESTID_PREFIX = "islock";
+static constexpr std::string_view INPUTLOCK_REQUESTID_PREFIX = "inlock";
```
Comment (kittywhiskers) [inline src/llmq/instantsend.cpp:30]: nit: switching to `constexpr string_view` should be a separate commit

File: src/llmq/init.cpp [hunk 5]
```diff
@@ -16,25 +16,27 @@
...
 
 CBLSWorker* blsWorker;
 
-void InitLLMQSystem(CEvoDB& evoDb, CTxMemPool& mempool, CConnman& connman, bool unitTests, bool fWipe)
+void InitLLMQSystem(NodeContext& node, CEvoDB& evoDb, CTxMemPool& mempool, CConnman& connman, bool unitTests, bool fWipe)
 {
     blsWorker = new CBLSWorker();
 
-    quorumDKGDebugManager = new CDKGDebugManager();
+    node.quorumDKGDebugManager = std::make_unique<CDKGDebugManager>();
```
Comment (kittywhiskers) [inline src/llmq/init.cpp:32]: Consider using `make_shared`/`make_shared` and migrate all `new` instantiations to it as a separate commit. Picking between either seems to involve a tradeoff between safety and performance. Thoughts @UdjinM6?

> The C++17 standard, while still not specifying the exact evaluation order for arguments, provides additional guarantees. All side effects of a function argument must be evaluated before the next argument is evaluated. This helps mitigate the risk in case of exceptions, but it is still preferable to use 'make_unique'.
>
> One thing should be noted about the 'make_shared' function. When it is used, the pointer's control block is allocated next to the managed object. This helps reduce the number of dynamic allocations and optimize the use of the CPU cache.

Source: https://pvs-studio.com/en/docs/warnings/v824/

File: src/llmq/dkgsessionmgr.h [hunk 6]
```diff
@@ -48,7 +49,7 @@ class CDKGSessionManager
     mutable std::map<ContributionsCacheKey, ContributionsCacheEntry> contributionsCache GUARDED_BY(contributionsCacheCs);
 
 public:
-    CDKGSessionManager(CConnman& _connman, CBLSWorker& _blsWorker, bool unitTests, bool fWipe);
+    CDKGSessionManager(CConnman& _connman, CBLSWorker& _blsWorker, CDKGDebugManager& dkgDebugMan, bool unitTests, bool fWipe);
```
Comment (kittywhiskers) [inline src/llmq/dkgsessionmgr.h:52]: nit: convention dictates name `_dkgDebugManager`

**Comment by UdjinM6** [comment]
pls see https://github.com/UdjinM6/dash/commit/0e3bda3c2825cff4ecd9afe6ccce86ca48d0528f
